networkx/utils/misc.py

```diff
@@ -227,22 +227,63 @@ def arbitrary_element(iterable):
     """Returns an arbitrary element of `iterable` without removing it.
 
     This is most useful for "peeking" at an arbitrary element of a set,
-    but can be used for any list, dictionary, etc., as well::
+    but can be used for any list, dictionary, etc., as well.
 
-        >>> arbitrary_element({3, 2, 1})
+    Parameters
+    ----------
+    iterable: `abc.collections.Iterable` instance
+        Any object that implements ``__iter__``, e.g. set, dict, list, tuple,
+        etc.
+
+    Returns
+    -------
+    The object that results from ``next(iter(iterable))``
+
+    Raises
+    ------
+    ValueError
+        If `iterable` is an iterator (because the current implementation of
+        this function would consume an element from the iterator).
+
+    Examples
+    --------
+    Arbitrary elements from common Iterable objects:
```

Comment:
Some code snippets are prefixed by `::`, others follow a `:` and are indented.  Which should it be, ideally?

Reply:
I've organized it so that code snippets that occur within the `Examples` section are not indented further (i.e. any explanatory text above them uses only one `:`), whereas code snippets outside of the examples section use the `::` so that they are properly identified as code blocks and highlighted etc.

I'm not sure what the right way is, but I personally like having the code examples within the `Examples` section non-indented.

Reply:
Great, that makes total sense!